source/lib/JMSLab/builders/build_lyx.py

```diff
@@ -26,19 +29,143 @@ def build_lyx(target, source, env):
         'exec_opts': '-E pdf2'
     }
     builder = LyxBuilder(target, source, env, **builder_attributes)
-    builder.execute_system_call()
+    builder.execute_system_call(target, env)
     return None
 
 
 class LyxBuilder(JMSLabBuilder):
-    '''
-    '''
+    
     def add_call_args(self):
-        '''
-        '''
         args = '%s %s %s > %s' % (os.path.normpath(self.target[0]),
                                   os.path.normpath(self.source_file),
                                   self.cl_arg,
                                   os.path.normpath(self.log_file))
         self.call_args = args
         return None
+
+
+    def create_handout(self):
+        '''
+        If beamer document class, convert Lyx notes to greyedout.
+        '''
+        
+        self.handout_out = str(self.main_target)
+        self.handout_in  = os.path.splitext(self.handout_out)[0] + '.lyx'
+
+        shutil.copy2(self.source_file, self.handout_in)
+        beamer = False
+        for line in fileinput.input(self.handout_in, inplace = True):
+            if r'\textclass beamer' in line:
+                beamer = True
+            elif r'\begin_inset Note Note' in line and beamer:
+                line = line.replace('Note Note', 'Note Greyedout')
+            print(line, end='')
+        
+        args = '%s %s %s > %s' % (self.handout_out,
+                                self.handout_in,
+                                self.cl_arg,
+                                os.path.normpath(self.log_file))
+        
+        self.handout_args = args
+        self.handout_call = '%s %s %s' % (self.executable, 
+                                          self.exec_opts, 
+                                          self.handout_args)
+
+        return None
+
+
+    def cleanup_handout(self):
+        '''
+        Copy handout pdf to desired locations.
+        Remove intermediate files.
+        '''
+        for x in self.handout_target_list:
+            shutil.copy2(self.handout_out, str(x))
+        os.remove(self.handout_out)
+        os.remove(self.handout_in)
+        return None
+
+
+    def do_call(self, target, env):
+        '''
+        Generate handout pdf if handout path exists in target and handout path
+        has the correct suffix and/or extension. Raise value error if intended 
+        behavior implied by target list contradicts behavior implied by 
+        HANDOUT_SFIX.
+        
+        Always generate main pdf.        
+        '''
+        
+        target_list = misc.make_list_if_string(target)
+        handout_sfix = '' if 'HANDOUT_SFIX' not in env else env['HANDOUT_SFIX']
```

Comment:
@rcalvo12 Ah, I see what you mean. I was trying to match the style of construction environment variables in Scons, [which](https://scons.org/doc/1.2.0/HTML/scons-user/a4774.html) are by convention in all caps, since `HANDOUT_SFIX` will be a construction variable. Do you think the extra convenience is enough that we should override the convention?

Reply:
@rcalvo12 I also like that the all caps makes it easier to tell in the python code which variables are from the environment and which are not. If we have
`handout_sfix = '' if 'handout_sfix' not in env else env['handout_sfix']`

then it might not be as obvious?

Reply:
@rcalvo12 @ew487 wanted to see if we can resolve this thread, thanks!

Reply:
Thanks @jmshapir. @ew487 I can see the argument you are making here for sticking to general scons convention. I think it's fine to leave as all-caps. It may be useful to create a brief section in the readme about the handout functionality with a specific code example so people have somewhere to reference if they do forget about the syntax here.

Reply:
Thanks @rcalvo12!

I'm in favor of keeping the main [readme](https://github.com/JMSLab/Template/tree/71ca49710dfed98e0461db45a39a85194dbf7e54#readme) pretty high-level so for now would be inclined to leave this detail in the docstring. If we later find it's confusing I'm glad to think about other places we could document it though.

@ew487 if this all seems good to you, feel free to resolve this thread, thanks!

Reply:
@rcalvo12 @jmshapir That sounds good to me! I will resolve the thread now.